### PR TITLE
Add integration test for CPU and Platform profiles

### PR DESCRIPTION
**Pull request by officialcjunior**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
* Add an integration test for CPU and Platform profiles.

* Return `NULL` if `target->platforms` is `NULL` in
`rz_arch_platform_target_new`

* Move the call of the API which adds the flags/comments for both
CPU and platform profiles to `rz_core_analysis_all()`.

**Test plan**

- [x] CI is green

**Closing issues**

None

### CODE REVIEW

**Comment by officialcjunior** [comment]
Right. The test I just introduced runs fine, and the other fails because the directory `.tmp/` where it stores the `.rzdb` is already there, I believe. 
Should I remove that check (at line 40) or get another directory?

**Comment by XVilka** [comment]
But it should be fine even if it exists, I think. See also related https://github.com/rizinorg/rizin/issues/1093

File: test/integration/test_cpu_platform_profiles.c [hunk 1]
```diff
@@ -0,0 +1,124 @@
+#include <rz_core.h>
+#include <rz_arch.h>
+#include <rz_project.h>
+
+#include "../unit/minunit.h"
+
+bool test_cpu_profiles() {
+
```
Comment (ret2libc) [inline test/integration/test_cpu_platform_profiles.c]:
```suggestion
```

File: test/integration/test_cpu_platform_profiles.c [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+	mu_assert_eq(item->offset, 0x00000004, "Flag DDRB not found");
+	item = rz_flag_get(core->flags, "PORTB");
+	mu_assert_eq(item->offset, 0x00000005, "Flag PORTB not found");
+
+	rz_core_free(core);
+	mu_end;
+}
+
+bool test_platform_profiles() {
+
```
Comment (ret2libc) [inline test/integration/test_cpu_platform_profiles.c]:
```suggestion
```

File: test/integration/test_cpu_platform_profiles.c [hunk 3]
```diff
@@ -0,0 +1,124 @@
+#include <rz_core.h>
```
Comment (ret2libc) [inline test/integration/test_cpu_platform_profiles.c:1]: @thestr4ng3r I probably missed the addition of this dir. How are we going to split things that need to go in test/integration and test/unit? I believe there are already some things in unit that are not that "unit". WDYT?

Comment (XVilka) [reply test/integration/test_cpu_platform_profiles.c:1]: Integration tests in this case are the ones that involve multiple parts/modules of Rizin, and usually represent some real use-case scenario.  They are good way to test possible use of the Rizin API.

File: test/integration/test_cpu_platform_profiles.c (hunk 2 above)
Comment (officialcjunior) [reply test/integration/test_cpu_platform_profiles.c]: Thank you!

File: test/integration/test_open_analyse_save_load_project.c [hunk 4]
```diff
@@ -37,7 +37,7 @@ bool test_open_analyse_save() {
 	size_t functions_count_expect = rz_list_length(functions);
 
 	// 4. Save into the project
-	if (!rz_file_exists(".tmp/")) {
+	if (!rz_file_is_directory(".tmp/")) {
```
Comment (wargio) [inline test/integration/test_open_analyse_save_load_project.c]: is this folder even legal for windows? @ret2libc

Comment (XVilka) [reply test/integration/test_open_analyse_save_load_project.c]: It is - test was passing on Windows

Comment (ret2libc) [reply test/integration/test_open_analyse_save_load_project.c]: Yeah I think it is nowadays.. anyway, fixed on another PR.

Comment (wargio) [inline test/integration/test_open_analyse_save_load_project.c]: You can remove this commit, since ret2libc has fixed this on another PR which is merged

Comment (officialcjunior) [reply test/integration/test_open_analyse_save_load_project.c]: Rebased to `dev`.

**Review by XVilka — APPROVED** [review]

File: test/integration/meson.build [hunk 5]
```diff
@@ -1,7 +1,8 @@
 if get_option('enable_tests')
   tests = [
     'open_analyse_save_load_project',
-    'bin_vfiles'
+    'bin_vfiles',
+    'cpu_platform_profiles'
```
Comment (ret2libc) [inline test/integration/meson.build]: Not important, but I think this should be valid in meson.build files. This way you don't have to modify the last line whenever you add a new entry to the array
```suggestion
    'cpu_platform_profiles',
```

**Review by ret2libc — APPROVED** [review]
Seems ok for what I understand.

On a different topic: I like the split in integration/unit tests! We could use the meson feature `suite: '` of `test` to split them and then run only unit by default.

**Pull request merged**